pkg/controller/cronjob/utils.go

```diff
@@ -234,6 +234,15 @@ func makeCreatedByRefJson(object runtime.Object) (string, error) {
 	return string(createdByRefJson), nil
 }
 
+func getFinishedStatus(j *batch.Job) (bool, batch.JobConditionType) {
+	for _, c := range j.Status.Conditions {
+		if (c.Type == batch.JobComplete || c.Type == batch.JobFailed) && c.Status == v1.ConditionTrue {
+			return true, c.Type
+		}
+	}
+	return false, ""
+}
```

Comment:
Can you combine `getFinishedStatus` with `IsJobFinished` in the following way:
```go
func IsJobFinished(j *batch.Job) bool {
	isFinished, _ = GetFinishedStatus(j)
	return isFinished
}
```
This will remove the code duplication. Btw. make the `GetFinishedStatus` public.